mdata/cache/ccache_metric.go

```diff
@@ -401,15 +415,22 @@ func (mc *CCacheMetric) Search(ctx context.Context, metric schema.AMKey, res *CC
 	}
 
 	if !res.Complete && res.From > res.Until {
-		log.Debug("CCacheMetric Search: Found from > until (%d/%d), printing chunks\n", res.From, res.Until)
-		mc.debugMetric()
+		log.Warn("CCacheMetric Search: Found from > until (%d/%d), key = %s, printing chunks\n", res.From, res.Until, mc.MKey.String())
+		mc.debugMetric(from-7200, until+7200)
+		res.Complete = false
+		res.Start = res.Start[:0]
+		res.End = res.End[:0]
+		res.From = from
+		res.Until = until
```

Comment:
so the idea here is that if the result is not complete then the cache search should return nothing at all, which will then cause the backend store to fetch everything that's been requested and overwrite what's in the cache (if there is anything there), which should then also heal potential corruptions, right? That's unfortunate if it's necessary to do that, but if it helps I guess it makes sense.

Reply:
If the result is simply not complete we still use the cache. If the result is incomplete but somehow from is after until, then we know something crazy happened. In practice, I don't see this happening a lot, but it does happen.